Source/NSMenu.m

```diff
@@ -1619,13 +1619,33 @@ + (void) popUpContextMenu: (NSMenu*)menu
 }
 
 + (void) popUpContextMenu: (NSMenu *)menu 
-                withEvent: (NSEvent *)event 
+                withEvent: (NSEvent *)event
                   forView: (NSView *)view 
                  withFont: (NSFont *)font
 {
   [menu _rightMouseDisplay: event];
 }
 
+- (void) popUpMenuPositionItem: (NSMenuItem *)item
+                    atLocation: (NSPoint) point
+                        inView: (NSView *) view
+{
+  NSRect cellFrame = [view convertRect: [view bounds] toView: nil];
+  NSWindow *w = [view window];
+  NSMenuView *mr = [self menuRepresentation];
+  NSUInteger selectedItem = [self indexOfItem: item];
+
+  cellFrame = [[view window] convertRectToScreen: cellFrame];
+  cellFrame.origin.x += point.x;
```

Comment:
Do you know in which coordinate system point is? We may have to fix this later on.

Reply:
I believe it is the point within the given view.  I will do more testing and fix in a later PR.